src/main/java/com/minecolonies/coremod/generation/SifterCraftingProvider.java

```diff
@@ -0,0 +1,364 @@
+                        .item(Items.COAL, 10)
+                        .item(Items.REDSTONE, 10)
+                        .build()),
+
+                new SifterMeshDetails(ModItems.sifterMeshIron, 4, new LootTableBuilder()
+                        .bonusRolls(0.025f)
+                        .empty(46)
+                        .item(Items.REDSTONE, 15)
+                        .item(Items.IRON_NUGGET, 15)
+                        .item(Items.COAL, 15)
+                        .item(Items.LAPIS_LAZULI, 5)
+                        .item(Items.IRON_INGOT, 1)
+                        .item(Items.GOLD_INGOT, 1)
+                        .item(Items.EMERALD, 1)
+                        .item(Items.DIAMOND, 1)
+                        .build()),
+
+                new SifterMeshDetails(ModItems.sifterMeshDiamond, 5, new LootTableBuilder()
+                        .bonusRolls(0.035f)
+                        .empty(40)
+                        .item(Items.REDSTONE, 20)
+                        .item(Items.IRON_NUGGET, 20)
+                        .item(Items.COAL, 20)
+                        .item(Items.LAPIS_LAZULI, 10)
+                        .item(Items.IRON_INGOT, 2)
+                        .item(Items.GOLD_INGOT, 2)
+                        .item(Items.EMERALD, 2)
+                        .item(Items.DIAMOND, 2)
+                        .build())
+
+                ).collect(Collectors.toList()));
+
+        inputs.put(Items.SAND, Stream.of(
+                new SifterMeshDetails(ModItems.sifterMeshString, 1, new LootTableBuilder()
+                        .empty(95)
+                        .item(Items.CACTUS, 2)
+                        .item(Items.SUGAR_CANE, 2)
+                        .build()),
+
+                new SifterMeshDetails(ModItems.sifterMeshFlint, 3, new LootTableBuilder()
+                        .empty(85)
+                        .item(Items.CACTUS, 5)
+                        .item(Items.SUGAR_CANE, 5)
+                        .item(Items.GOLD_NUGGET, 5)
+                        .build()),
+
+                new SifterMeshDetails(ModItems.sifterMeshIron, 4, new LootTableBuilder()
+                        .bonusRolls(0.025f)
+                        .empty(60)
+                        .item(Items.CACTUS, 10)
+                        .item(Items.SUGAR_CANE, 10)
+                        .item(Items.GOLD_NUGGET, 10)
+                        .item(Items.COCOA_BEANS, 10)
+                        .build()),
+
+                new SifterMeshDetails(ModItems.sifterMeshDiamond, 5, new LootTableBuilder()
+                        .bonusRolls(0.035f)
+                        .empty(40)
+                        .item(Items.CACTUS, 15)
+                        .item(Items.SUGAR_CANE, 15)
+                        .item(Items.GOLD_NUGGET, 15)
+                        .item(Items.COCOA_BEANS, 15)
+                        .build())
+
+                ).collect(Collectors.toList()));
+
+        inputs.put(Items.SOUL_SAND, Stream.of(

```

Comment:
Did we also want to add SOUL_SOIL?

Reply:
I did have that in my previous PR, but you didn't in yours, so I've left it out for the moment.  It can be in a follow-up.

I previously had it using the soul sand table (which was just a one-liner at the time, but in this version you'd have to duplicate the whole code block and it'd create separate tables); and on further reflection it might be better giving it a different loot table anyway, perhaps something more like the dirt table but dropping nether saplings?  Something for you to think about, anyway.  😄

Reply:
Some of the other weights might need a little tweaking after further experimentation.  I suspect that if you do any dirt sifting you're going to be drowning in wheat seeds before too long.  There's only so much the farmer and composter can take. 🤣